src/main/java/net/mcreator/ui/modgui/codeviewer/ModElementCodeViewer.java

```diff
@@ -110,6 +109,7 @@ private synchronized void reload() {
 			new Thread(() -> {
 				updateRunning = true;
 				try {
+					Thread.sleep(50); // resolves freezing if previous updates are not complete yet

```

Comment:
If previous update running is the problem, you should use lock and synchronize the process, not simply do sleeping

Reply:
Are you talking about the thread itself or about methods called in it (e. g. the ones that update code viewer contents)? Also, what should I lock?

Reply:
Let's see if I understood you or not.

Reply:
You wrote:

> resolves freezing if previous updates are not complete yet

so instead of sleeping, wait until

> previous updates are not complete

Reply:
Complete?
So I need to change it again?

Reply:
You should not use sleep for this

Reply:
I was not able to use locks, instead I decided to use `ThreadPoolExecutor` which actually worked, but now I'm going with this:
https://github.com/MCreator/MCreator/blob/a14e84f82234f3a9630a70112713a75554ff3e5d/src/main/java/net/mcreator/ui/modgui/codeviewer/ModElementCodeViewer.java#L200

Reply:
This is same as sleep, it will timeout on 50ms

Reply:
>Removing thread there will make reload run on UI thread which is not ok

@KlemenDEV But as long as this `Runnable` is surrounded by thread constructor, the _synchronized_ method modifier does not apply to the task inside the thread, in this case it just means **only 1 _thread_ can be started at a time**.
So, why is it not OKay?

Reply:
It's finally resolved 🎉